namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java

```diff
@@ -334,28 +328,42 @@ public RegisterBrokerResult registerBroker(
-                            final TopicConfig topicConfig = entry.getValue();
-                            if (isPrimeSlave) {
+                    if (registerFirst || this.isBrokerTopicConfigChanged(clusterName, brokerAddr, topicConfigWrapper.getDataVersion())) {
+                        if (isPrimeSlave) {
+                            for (TopicConfig topicConfig : tcTable.values()) {
                                 // Wipe write perm for prime slave
                                 topicConfig.setPerm(topicConfig.getPerm() & (~PermName.PERM_WRITE));
+                                this.createAndUpdateQueueData(brokerName, topicConfig);
+                            }
+                        } else {
+                            for (TopicConfig topicConfig : tcTable.values()) {
+                                this.createAndUpdateQueueData(brokerName, topicConfig);
                             }
-                            this.createAndUpdateQueueData(brokerName, topicConfig);
                         }
-                    }
 
-                    if (this.isBrokerTopicConfigChanged(clusterName, brokerAddr, topicConfigWrapper.getDataVersion()) || registerFirst) {
                         //the topicQueueMappingInfoMap should never be null, but can be empty
                         for (Map.Entry<String, TopicQueueMappingInfo> entry : topicQueueMappingInfoMap.entrySet()) {
-                            if (!topicQueueMappingInfoTable.containsKey(entry.getKey())) {
-                                topicQueueMappingInfoTable.put(entry.getKey(), new HashMap<>());
-                            }
+                            Map<String, TopicQueueMappingInfo> topicQueueMappingInfo = ConcurrentHashMapUtils.computeIfAbsent((ConcurrentHashMap<String, Map<String, TopicQueueMappingInfo>>) topicQueueMappingInfoTable, entry.getKey(), k -> new HashMap<>());
                             //Note asset brokerName equal entry.getValue().getBname()
                             //here use the mappingDetail.bname
-                            topicQueueMappingInfoTable.get(entry.getKey()).put(entry.getValue().getBname(), entry.getValue());
+                            TopicQueueMappingInfo value = entry.getValue();
+                            topicQueueMappingInfo.put(value.getBname(), value);
+                        }
+                    } else {
+                        if (isPrimeSlave) {
+                            for (TopicConfig topicConfig : tcTable.values()) {
+                                if (this.isTopicConfigChanged(false, brokerName, topicConfig.getTopicName())) {
+                                    // Wipe write perm for prime slave
+                                    topicConfig.setPerm(topicConfig.getPerm() & (~PermName.PERM_WRITE));
+                                    this.createAndUpdateQueueData(brokerName, topicConfig);
+                                }
+                            }
+                        } else {
```

Comment:
suggest not using too much if..else.., replace with just `if` clause

Reply:
Moving `if` unrelated to the loop outside of the loop will result in better performance